### PR TITLE
enable input tokens in magic commands

### PR DESCRIPTION
**Pull request by jonsequitur**
This will address #1554.

image

image

### CODE REVIEW

**Review by colombod — COMMENTED** [review]
So far it looks good

**Review by colombod — APPROVED** [review]

File: src/Microsoft.DotNet.Interactive.FSharp/FSharpKernel.fs [hunk 1]
```diff
@@ -412,16 +412,16 @@ type FSharpKernel () as this =
...
 
     interface IKernelCommandHandler<RequestHoverText> with
         member this.HandleAsync(command: RequestHoverText, context: KernelInvocationContext) = handleRequestHoverText command context |> Async.StartAsTask :> Task
 
     interface IKernelCommandHandler<SubmitCode> with
-        member this.HandleAsync(command: SubmitCode, context: KernelInvocationContext) = handleSubmitCode command context |> Async.StartAsTask :> Task
+        member this.HandleAsync(command: SubmitCode, context: KernelInvocationContext) = handleSubmitCode command context
 
     interface IKernelCommandHandler<ChangeWorkingDirectory> with
         member this.HandleAsync(command: ChangeWorkingDirectory, context: KernelInvocationContext) = handleChangeWorkingDirectory command context |> Async.StartAsTask :> Task
```
Comment (colombod) [inline src/Microsoft.DotNet.Interactive.FSharp/FSharpKernel.fs:427]: should we complete the migration to task ?

File: src/Microsoft.DotNet.Interactive/Connection/KernelCommandAndEventTextStreamReceiver.cs [hunk 2]
```diff
@@ -5,23 +5,32 @@
...
 {
-    public class KernelCommandAndEventTextReaderReceiver : KernelCommandAndEventDeserializingReceiverBase
+    public class KernelCommandAndEventTextStreamReceiver : KernelCommandAndEventDeserializingReceiverBase
     {
         private readonly TextReader _reader;
+        private readonly string _name;
 
-        public KernelCommandAndEventTextReaderReceiver(TextReader reader)
+        public KernelCommandAndEventTextStreamReceiver(
+            TextReader reader,
```
Comment (colombod) [inline src/Microsoft.DotNet.Interactive/Connection/KernelCommandAndEventTextStreamReceiver.cs:19]: Why rename to TextStream if the input is still TextReader?

File: src/Microsoft.DotNet.Interactive/Connection/KernelCommandAndEventTextStreamSender.cs [hunk 3]
```diff
@@ -7,28 +7,32 @@
...
 namespace Microsoft.DotNet.Interactive.Connection
 {
     public class KernelCommandAndEventTextStreamSender : IKernelCommandAndEventSender
     {
-        public Uri RemoteHostUri { get; }
         private readonly TextWriter _writer;
 
-        public KernelCommandAndEventTextStreamSender(TextWriter writer, Uri remoteHostUri)
+        public KernelCommandAndEventTextStreamSender(
+            TextWriter writer,
```
Comment (colombod) [inline src/Microsoft.DotNet.Interactive/Connection/KernelCommandAndEventTextStreamSender.cs:18]: Same question here sender using a TextWriter but having TextStream in the name

File: src/Microsoft.DotNet.Interactive/KernelInvocationContext.cs [hunk 4]
```diff
@@ -150,15 +150,9 @@ internal void Cancel()
...
         {
-            try
-            {
-                if (!_cancellationTokenSource.IsCancellationRequested)
-                {
-                    _cancellationTokenSource.Cancel();
-                }
-            }
-            catch (ObjectDisposedException)
+            if (!_cancellationTokenSource.IsCancellationRequested)
```
Comment (colombod) [inline src/Microsoft.DotNet.Interactive/KernelInvocationContext.cs:153]: we don't need this at all?

File: src/Microsoft.DotNet.Interactive.FSharp/FSharpKernel.fs (hunk 1 above)
Comment (jonsequitur) [reply src/Microsoft.DotNet.Interactive.FSharp/FSharpKernel.fs:427]: Probably. At this point I'm stabbing to figure out why one of a few different random tests keep failing in CI.

**Pull request merged**